Jenkinsfile

```diff
@@ -1,38 +1,31 @@
 pipeline {
-    agent none
+    agent { label 'dockerhost' }
     environment {
         CI = 'true'
+        DOCKER_REPO_NAME = "kiva/kiosa-dev"
         TAG_NAME = "${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
+        TAGGED_IMAGE_NAME = "${env.DOCKER_REPO_NAME}:${env.TAG_NAME}"
     }
     stages {
-        stage('Test') {
-            agent {
-                dockerfile {
-                    args '-v $HOME/.gradle:/root/.gradle'
-                }
-            }
+        stage('Build and test') {
             steps {
-                echo 'Testing...'
-                sh "./gradlew build"
+                withDockerContainer(image: 'openjdk:8-jdk-alpine', args: '-v $HOME/.gradle:/root/.gradle') {
+                    sh "./gradlew build"
+                }
             }
         }
-        stage('Build') {
-            agent { label 'dockerhost' }
+        stage('Build production image') {
             steps {
-                echo 'Building...'
-                sh "docker build -t kiva/kiosa-dev:${TAG_NAME} -f Dockerfile.prod ."
+                sh "cp build/libs/*.jar app.jar"
```

Comment:
Jenkins mounted the workspace to the previous container and set the working directory there, so the built jar is directly available in our current workspace. We use the cp with wildcard so we don't have to extract the actual jar name from the folder.